node/node.go

```diff
@@ -504,6 +506,11 @@ func (n *Node) ConfigureRPC() (*rpccore.Environment, error) {
 	if pubKey == nil || err != nil {
 		return nil, fmt.Errorf("can't get pubkey: %w", err)
 	}
+
+	genDoc, err := n.genesisDocProvider()
```

Comment:
I would make genesis time a construction-time parameter for this object type, and pass it down from where we have obtained the genesis doc in setup.

Reply:
So for the RPC, I am not so sure about removing it without measuring the impact on performance given that we do not know 100% how much this change affects the users. The changes in the node are, at least from a Comet point of view without consequences. If someone is using this getter to get the genesis chunk and requires performance, they can very well cache the file on their end .

Reply:
Agree, let's leave the logic for chunks outside the scope here. During my investigations the chunks get allocated on the head and contribute to the process memory allocation. One solution would be to allocate memory on a chunk request. Like if a user requests chunk 1 then you cache it (maybe seeking the file instead of reading the whole file). But again this is a much bigger scope for now. I'll mark this one as resolved.